src/Microsoft.Data.SqlClient/src/Microsoft/Data/SqlClient/SqlConnectionPoolKey.cs

```diff
@@ -102,18 +111,19 @@ internal SqlClientOriginalNetworkAddressInfo OriginalNetworkAddressInfo
                 return _originalNetworkAddressInfo;
             }
         }
+#endif
 
         public override bool Equals(object obj)
         {
-            SqlConnectionPoolKey key = obj as SqlConnectionPoolKey;
-
-            return (key != null &&
-                    _credential == key._credential &&
-                    ConnectionString == key.ConnectionString &&
-                    string.CompareOrdinal(_accessToken, key._accessToken) == 0 &&
-                    _serverCertificateValidationCallback == key._serverCertificateValidationCallback &&
-                    _clientCertificateRetrievalCallback == key._clientCertificateRetrievalCallback &&
-                    _originalNetworkAddressInfo == key._originalNetworkAddressInfo);
+            return (obj is SqlConnectionPoolKey key
+                && _credential == key._credential
+                && ConnectionString == key.ConnectionString
+#if NETFRAMEWORK
+                && _serverCertificateValidationCallback == key._serverCertificateValidationCallback
+                && _clientCertificateRetrievalCallback == key._clientCertificateRetrievalCallback
+                && _originalNetworkAddressInfo == key._originalNetworkAddressInfo
+#endif
+                && string.CompareOrdinal(_accessToken, key._accessToken) == 0);
```

Comment:
This last condition should come before the #if block?

Reply:
Thanks! I've moved the access token check to line 121 before the ifdef block in the next commit.